force-app/main/default/classes/HH_Container_TEST.cls

```diff
@@ -662,6 +662,53 @@ private class HH_Container_TEST {
         System.assertEquals(2, records.size(), 'Expected records when Contact.Name is encrypted and the search query is valid: ' + records);
     }
 
+    /*********************************************************************************************************
+    * @description Verifies required methods can be run as a User without Undeliverable Address permissions
+    */
+    public static testMethod void testPageLoadsWithoutUndeliverableAddressPermissions() {
+        User adminUser = UTIL_UnitTestData_TEST.createUserWithoutInsert(UTIL_Profile.SYSTEM_ADMINISTRATOR);
+        Id adminProfileId = [SELECT Id FROM Profile WHERE Name = :UTIL_Profile.SYSTEM_ADMINISTRATOR LIMIT 1].Id;
+        
+        String addressUndeliverableField = 
+            String.valueOf(Address__c.SObjectType) + '.' + String.valueOf(Address__c.Undeliverable__c);
+        String contactUndeliverableField = 
+            String.valueOf(Contact.SObjectType) + '.' + String.valueOf(Contact.Undeliverable_Address__c);
+        String accountUndeliverableField = 
+            String.valueOf(Account.SObjectType) + '.' + String.valueOf(Account.Undeliverable_Address__c);
+        List<FieldPermissions> fieldPermissions = [
+            SELECT Id, PermissionsRead, PermissionsEdit
+            FROM FieldPermissions
+            WHERE Parent.ProfileId = :adminProfileId
+            AND (Field = :addressUndeliverableField
+                OR Field =: contactUndeliverableField
+                OR Field = : accountUndeliverableField
+            )
+        ];
+        for(FieldPermissions fieldPermission : fieldPermissions){
```

Comment:
Is it worth testing some other combinations, such as:
- Edit Access to the Address.Undeliverable field, but only Read to the Account/Contact fields
- Read Access to the Address.Undeliverable field, but no to the Account/Contact fields
- No Access to the Address.Undeliverable field, but Edit to the Account/Contact fields

There's obviously a lot of permutations. The question is how many of them we want to test to be sure this isn't going to still generate unexpected errors for customers.

Reply:
I added one additional test that retains Account.Undeliverable access, since I needed to query that field as part of the test, I'm not sure testing every combination is worth it here, unless we find issues.

Reply:
We're more concerned with data integrity than unexpected errors in this situation. The problem we were seeing is that the Undeliverable field would get set to an unexpected value if you didn't have permissions to any of the fields, because we relied on field visibility to drive the automated updates.